packages/atomic/src/components/commerce/atomic-product-image/atomic-product-image.ts

```diff
@@ -0,0 +1,286 @@
+ * @part active-indicator - The currently active indicator in the carousel.
+ */
+@customElement('atomic-product-image')
+@bindings()
+@withTailwindStyles
+export class AtomicProductImage
+  extends LitElement
+  implements InitializableComponent<Bindings>
+{
+  @state() public bindings!: Bindings;
+
+  public productController = createProductContextController(this);
+
+  @state() private product!: Product;
+
+  @state() public error!: Error;
+  @state() private useFallback = false;
+  @state() private currentImage = 0;
+
+  public initialize() {
+    if (!this.product && this.productController.item) {
+      this.product = this.productController.item;
+    }
+  }
+
+  /**
+   * The product field which the component should use. This will look for the field in the product object first, then in the product.additionalFields object.
+   */
+  @property({type: String, reflect: true}) field: string = 'ec_thumbnails';
+
+  /**
+   * The product field that contains the alt text for the images. This will look for the field in the product object first, then in the product.additionalFields object.
+   * If the product has multiple images, the value of the `imageAltField` will be used as the alt text for every image.
+   *
+   * If the field is not specified, or does not contain a valid value, the alt text will be set to "Image {index} out of {totalImages} for {productName}".
+   */
+  @property({type: String, reflect: true, attribute: 'image-alt-field'})
+  imageAltField?: string;
+
+  /**
+   * An fallback image URL that will be used in case the specified image is not available or an error is encountered.
+   */
+  @property({type: String, reflect: true}) fallback: string =
+    // eslint-disable-next-line @cspell/spellchecker
+    'data:image/svg+xml,%3Csvg xmlns="http://www.w3.org/2000/svg" width="50" height="50" viewBox="0 0 50 50"%3E%3Crect width="50" height="50" fill="none" stroke="none"%3E%3C/rect%3E%3C/svg%3E';
+
+  /**
+   * Navigates to the previous image in the carousel.
+   */
+  public async previousImage() {
+    this.currentImage =
+      this.currentImage - 1 < 0
+        ? this.numberOfImages - 1
+        : this.currentImage - 1;
+  }
+
+  /**
+   * Navigates to the next image in the carousel.
+   */
+  public async nextImage() {
+    this.currentImage = (this.currentImage + 1) % this.numberOfImages;
+  }
+
+  /**
+   * Navigates to the specified image index.
+   *
+   * @param index - The index of the image to navigate to.
+   */
+  public async navigateToImage(index: number) {
+    this.currentImage = index;
+  }
+
+  private logWarning(message: string) {
+    this.bindings.engine.logger.warn(message, this);
+  }
+
+  private handleImageError(event: Event) {
+    const image = event.target as HTMLImageElement;
+    const message = `The image url "${image.src}" is not valid or could not be loaded. You might want to add a "fallback" property.`;
+    if (this.fallback && image.src !== this.fallback) {
+      this.useFallback = true;
```

Comment:
Since `this.useFallback` is a state property, I believe setting it to `true` here would trigger a re-render. This would imply that setting `image.src` to `this.fallback` would be redundant as we'll be doing `const finalUrl = this.useFallback ? this.fallback : url;` in the `images` getter that will get called in the `render` function.